python/lsst/sims/movingObjects/linearObs.py

```diff
@@ -0,0 +1,293 @@
+
+
+    def generateEphs(self, sso):
+        """Generate ephemerides for all times in self.ephTimes.
+        """
+        self.pyephem.setOrbits(sso)
+        oorbEphs = self.pyephem._generateOorbEphs(self.ephTimes, obscode=self.obscode)
+        ephs = self.pyephem._convertOorbEphs(oorbEphs, byObject=True)
+        return ephs
+
+    # Linear interpolation
+    def interpolateEphs(self, ephs, i=0):
+        """
+        Generate linear interpolations between the quantities in ephs over time.
+        """
+        interpfuncs = {}
+        for n in ephs.dtype.names:
+            if n == 'time':
+                continue
+            interpfuncs[n] = interpolate.interp1d(ephs['time'][i], ephs[n][i], kind='linear',
+                                                  assume_sorted=True, copy=False)
+        return interpfuncs
+
+    def _setupCamera(self):
+        """
+        Initialize camera mapper, etc.
+        """
+        self.mapper = LsstSimMapper()
+        self.camera = self.mapper.camera
+        self.epoch = 2000.0
+        self.cameraFov = np.radians(2.1)
+
+    def ssoInFov(self, interpfuncs, simdata, rFov=np.radians(1.75),
+                 useCamera=True,
+                 simdataRaCol = 'fieldRA', simdataDecCol='fieldDec'):
+        """
+        Return the indexes of the simdata observations where the object was inside the fov.
+        """
+        # See if the object is within 'rFov' of the center of the boresight.
+        raSso = np.radians(interpfuncs['ra'](simdata['expMJD']))
+        decSso = np.radians(interpfuncs['dec'](simdata['expMJD']))
+        sep = haversine(raSso, decSso, simdata[simdataRaCol], simdata[simdataDecCol])
+        if not useCamera:
+            idxObsRough = np.where(sep<rFov)[0]
+            return idxObsRough
+        # Or go on and use the camera footprint.
+        try:
+            self.camera
+        except AttributeError:
+            self._setupCamera()
+        idxObs = []
+        idxObsRough = np.where(sep<self.cameraFov)[0]
+        for idx in idxObsRough:
+            mjd_date = simdata[idx]['expMJD']
+            mjd = ModifiedJulianDate(TAI=mjd_date)
+            obs_metadata = ObservationMetaData(pointingRA=np.degrees(simdata[idx][simdataRaCol]),
+                                               pointingDec=np.degrees(simdata[idx][simdataDecCol]),
+                                               rotSkyPos=np.degrees(simdata[idx]['rotSkyPos']),
+                                               mjd=mjd)
+            raObj = np.array([interpfuncs['ra'](simdata[idx]['expMJD'])])
+            decObj = np.array([interpfuncs['dec'](simdata[idx]['expMJD'])])
+            # Catch the warnings from astropy about the time being in the future.
+            with warnings.catch_warnings(record=False):
+                warnings.simplefilter('ignore')
+                chipNames = chipNameFromRaDec(ra=raObj,dec=decObj, epoch=self.epoch,
```

Comment:
Awesome. See, this (and the above comment about AngularSeparation) are excellent examples of why we should have reviews, even above and beyond the whole "catching errors" and "making code more readable" and "catching silly errors".